great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -258,9 +264,9 @@ def get_batch_list_from_batch_request(
                 )
             # Creating the batch_spec is our hook into the execution engine.
             batch_spec = SqlAlchemyDatasourceBatchSpec(**batch_spec_kwargs)
-            data, markers = self.datasource.execution_engine.get_batch_data_and_markers(
-                batch_spec=batch_spec
-            )
+            data, markers = SqlAlchemyExecutionEngine(
```

Comment:
We discussed leaving the `execution_engine` on the `datasource` as a property that is not created on `__init__` but instead is deferred until referenced on the property.